apps/system/js/app_chrome.js

```diff
@@ -830,7 +830,10 @@
           this.expand();
         }
         this.scrollable.scrollTop = 0;
-        this.pinned = false;
+        // TODO: Check if the new location is pinned
+        // Taking it from the initial value for now
+        var chromeConfig = this.app.config && this.app.config.chrome;
+        this.pinned = chromeConfig && chromeConfig.pinned;
```

Comment:
So then if I open a pinned site, and click on any link, it gets expanded? This was implemented in this bug, mainly to make it more testable, so I guess it makes sense to leave it to bug 1168962. I wouldn't mind to leave it and make the 'pinned' feature in bookmarks dependent on the pinning_the_web_enabled setting, though.